File: Table_Hachage/hachage.c
#include "Hachage.h"

int	main(int ac, char **av)
{
	int i;
	int choice;
	char content[256];

	i = 1;
	while (i < ac)
	{
		addHachage(av[i]);
		i++;
	}
	while (1)
	{
		printf("\n===== HASH TABLE =====\n");
		printf("1. Display\n");
		printf("2. Add\n");
		printf("3. Delete\n");
		printf("0. Quit\n");
		printf("Choice: ");
		scanf("%d", &choice);

		switch (choice)
		{
            case 1:
                printTable();
                break ;
            case 2:
                printf("Word: ");
                scanf("%255s", content);
                addHachage(content);
                break ;
            case 3:
                printf("Word: ");
                scanf("%255s", content);
                deleteHachage(content);
                break ;
            case 0:
                free_table();
                return (0);
            default:
                printf("Invalid choice.\n");
		}
	}

	return (0);
}

File: Table_Hachage_py/hachage.py
import sys
import subprocess
import time
from HashTable import HashTable


def clear_screen() -> None:
    subprocess.run(["clear"])


def main(args: list[str]) -> None:
    table = HashTable()

    for word in args:
        table.add(word)

    while True:
        clear_screen()

        print("====== HASH TABLE ======")
        print("1. Display")
        print("2. Add")
        print("3. Delete")
        print("0. Quit")

        choice = input("\nChoice: ").strip()

        if choice == "1":
            table.display()
            input("\nPress Enter to continue...")

        elif choice == "2":
            word = input("Word: ").strip()
            table.add(word)
            input("\nWord added. Press Enter to continue...")

        elif choice == "3":
            word = input("Word: ").strip()
            table.delete(word)
            input("\nPress Enter to continue...")

        elif choice == "0":
            clear_screen()
            print("👋 Bye!")
            time.sleep(2)
            clear_screen()
            sys.exit()

        else:
            print("\n❌ Invalid choice.", file=sys.stderr)
            input("Press Enter to continue...")


if __name__ == "__main__":
    try:
        main(sys.argv[1:])
    except Exception:
        print("Error")
        sys.exit()
    except KeyboardInterrupt:
        print("Keyboard interrupt")
        sys.exit()
